ERS stack processing

yjzhenglamarmota opened the pull request:
I have been processing ERS1/2 data from ESA (http://esar-ds.eo.esa.int/socat/SAR_IMS_1P/) and thought it would be helpful to add the following script to stripmapStack before we can run stackStripMap.py for ERS data.

unpackFrame_ERS_ENV.py

I have also updated ODR.py because the updated Delft orbits have some format changes, and the original hard-coded line https://github.com/yjzhenglamarmota/isce2/blob/7fcf2c5662627842a788e5954778d98e2b3ca3e7/components/isceobj/Orbit/ODR.py#L185parser no longer works. The updated version work for both the old arclist files and the new ones.

yunjunz committed:
bugfix in `cuDenseOffsets.py --full/out-geom` for topsApp product

Fix a bug while running `cuDenseOffsets.py --full/out-geom` on topsApp.py products by:
+ setting `lat` as the first geometry dataset, instead of `hgt`, because the former is shared between topsApp and topsStack products
+ add `z` from topsApp, in addition to `hgt` from topsStack, as the input geometry dataset

MYIP003 committed:
Bug on alosStack

The issue was opened by Dani-Lindsay, at first cmd_3.sh fails during alosStack/ion_check.py because  imagemagick was not installed. After installing imagemagick by conda install -c conda-forge imagemagick, I got another error message: 

convert-im6.q16: unable to read font `helvetica' @ error/annotate.c/RenderFreetype/1396.

The error message can be fixed by changing line 106 in ion_check.py from
runCmd("montage -pointsize {} -label 'original' {} -label 'ionosphere' {} -label 'corrected' {} -geometry +{} -compress LZW{} {}.tif".format(
to 
runCmd("montage -font DejaVu-Sans -pointsize {} -label 'original' {} -label 'ionosphere' {} -label 'corrected' {} -geometry +{} -compress LZW{} {}.tif".format(

yjzhenglamarmota committed:
adding scripts -preparing ERS for stack processing

yjzhenglamarmota committed:
adding scripts -preparing ERS for stack processing

yjzhenglamarmota committed:
adding scripts -preparing ERS for stack processing

yjzhenglamarmota committed:
adding scripts -preparing ERS for stack processing

yjzhenglamarmota committed:
Combine prepareERS_ENVstack,py with unpackFrame_ERS_ENV.py in the latter.

yjzhenglamarmota committed:
Combine prepareERS_ENVstack,py with unpackFrame_ERS_ENV.py in the latter.

yjzhenglamarmota committed:
Combine prepareERS_ENVstack,py with unpackFrame_ERS_ENV.py in the latter.

yjzhenglamarmota committed:
Combine prepareERS_ENVstack,py with unpackFrame_ERS_ENV.py in the latter.

yjzhenglamarmota committed:
Fix bugs

yjzhenglamarmota committed:
Fix bugs

bjmarfito committed:
Added gdal2isce_xml when running multilook gdal

yjzhenglamarmota committed:
Add geolocation constraint option

yjzhenglamarmota committed:
Add geolocation constraint option

yjzhenglamarmota committed:
Update unpackFrame_ERS_ENV.py

Added orbit data type

yjzhenglamarmota committed:
Update unpackFrame_ERS_ENV.py

Added orbit data type

yjzhenglamarmota committed:
Update ODR.py

The arclist file in the more complete ODR orbit files downloaded from http://www.deos.tudelft.nl/AS/pieter/Local/ERS/index.html (contains ERS-2 orbit files beyond 2003) has slight format change and the original hard coded line parser no longer works. The updated version should work both for the new and old versions of arclist.

yjzhenglamarmota committed:
Update ODR.py

The arclist file in the more complete ODR orbit files downloaded from http://www.deos.tudelft.nl/AS/pieter/Local/ERS/index.html (contains ERS-2 orbit files beyond 2003) has slight format change and the original hard coded line parser no longer works. The updated version should work both for the new and old versions of arclist.

EJFielding committed:
Added "char" type in ALOS_fbd2fbsmodule.c and ALOS_fbs2fbdmodule.c (#815)

vincentschut committed:
add symlink in dockerfile

rtburns-jpl committed:
Merge pull request #823 from vincentschut/add-isce-to-isce2-symlink-in-docker

add symlink in dockerfile

yunjunz committed:
alosStack: ignore non-numeric folders in the data/download dir

+ ignore the non-numeric foldres in the data download directory, similar to stripmapStack, in the following scripts:
   - StackPublic
   - create_cmds
   - read_data

+ create_cmds: print out the number of acquisitions and pairs

+ create_cmds: ignore empty input of `insar.bbox` arg, in addition to the existing None input, otherwise, it causes the following err

```
Traceback (most recent call last):
  File "/home/yunjunz/tools/isce2/src/isce2/contrib/stack/alosStack/geocode.py", line 72, in <module>
    bbox = [float(x) for x in bbox.split('/')]
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/yunjunz/tools/isce2/src/isce2/contrib/stack/alosStack/geocode.py", line 72, in <listcomp>
    bbox = [float(x) for x in bbox.split('/')]
            ^^^^^^^^
ValueError: could not convert string to float: ''
```

yunjunz committed:
prepSlcALOS2: support alosStack convention via --alosStack option

rtburns-jpl committed:
Merge pull request #687 from MYIP003/patch-1

`alosStack/ion_check`: specify font for while calling `montage`

hfattahi commented on contrib/stack/stripmapStack/unpackFrame_ERS_ENV.py:
```diff
@@ -0,0 +1,119 @@
+    obj.frame.getImage().renderHdr()
+    obj.extractDoppler()
+
+    # ####  computation of "poly" adapted from line 339 - line 353 of  Components/isceobj/Sensor/ERS_EnviSAT_SLC.py ###
+    #######  removed this section and added  obj.extractDoppler() above instead. Doesn't seem to change anything in the processing. The latter is required for cropFrame.
+    # coeffs = obj.dopplerRangeTime
+    # dr = obj.frame.getInstrument().getRangePixelSize()
+    # rref = 0.5 * Const.c * obj.rangeRefTime
+    # r0 = obj.frame.getStartingRange()
+    # norm = 0.5 * Const.c / dr
+
+    # dcoeffs = []
+    # for ind, val in enumerate(coeffs):
+    #     dcoeffs.append( val / (norm**ind))
+    
+    # poly = Poly1D.Poly1D()
+    # poly.initPoly(order=len(coeffs)-1)
+    # poly.setMean( (rref - r0)/dr - 1.0)
+    # poly.setCoeffs(dcoeffs)
```
please remove commented code.

yjzhenglamarmota replied:
Done.

hfattahi commented on components/isceobj/Orbit/ODR.py:
```diff
@@ -184,15 +184,17 @@ def parse(self):
+        arc.number = line.split()[0] # Arc number
+        arc.start = datetime.datetime.strptime(" ".join(line.split()[1:3]),'%y%m%d %H:%M') # Starting time for the arc
+        arc.stop = datetime.datetime.strptime(" ".join(line.split()[4:6]),'%y%m%d %H:%M') # End time for the arc
+        ###### commented the following because it is not always present in the new arclist file ############
+        # arc.slrResidual = line[34:38] # Satellite laser ranging residual in cm        
+        # arc.crossOver = line[39:43]
+        # arc.altimeter = line[45:49]
+        # arc.repeat = line[51:57] # Repeat cycle in days
+        # arc.version = line[58:61] # Version number
```
Please remove the commented code.

yjzhenglamarmota replied:
Done.

hfattahi reviewed (commented):
Thanks @yjzhenglamarmota. The changes look fine to me. Would you please just remove the commented codes. Also would you please confirm this is the latest and greatest that worked for you in processing the large stack of ERS data?

rtburns-jpl committed:
Merge pull request #834 from yunjunz/alos_stack

`alosStack`: easy data preparation + ignore non-numeric folders in the data/download dir

rtburns-jpl committed:
Merge pull request #703 from bjmarfito/patch-1

Added gdal2isce_xml when running multilook gdal in mergeBursts.py

rtburns-jpl committed:
Update fetchOrbit.py for new Copernicus data source (#801)

* Update fetchOrbit.py for new Copernicus data source

* Move username/password assignment outside loop

Avoid reentering when downloading multiple files

* Use persistent copernicus auth token saved to file

Co-authored-by: Eric Lindsey <ericlindsey@users.noreply.github.com>

* Read username/password from .netrc file when needed

---------

Co-authored-by: Eric Lindsey <ericlindsey@users.noreply.github.com>

yunjunz committed:
docs: always use 4-digit frame number for alos2 (#708)

bjmarfito committed:
Process Lutan-1 stripmap SLC images using stripmapApp.py (#852)

* Update estimateIono.py

* Update runDispersive.py

* Added Lutan-1 support

* Added revision for Lutan-1

* Added Lutan 1 in SConscript and CMakeLists.txt

* Fixed Lutan-1 name

* Bug fixes for Lutan-1 reader

* Added 128k on the Fortran code

* Added the ability to ingest annotation file orbit info

* Added warning for using orbit info from annotation file

* Update README.md

---------

Co-authored-by: bjmarfito <bjmarfito@github.com>
Co-authored-by: Zhang Yunjun <yunjunz@outlook.com>

CunrenLiang committed:
removing azimuth burst ramps caused by ionosphere in topsStack (#600)

* removing azimuth burst ramps caused by ionosphere in topsStack

These updates are for removing azimuth burst ramps caused by ionosphere in the ionospheric correction in topsStack. The updates are only within this module.

* Update Stack.py

* update argument for s1_select_ion.py

* Update runFrameOffset.py

* Update contrib/stack/topsStack/README.md

Co-authored-by: Zhang Yunjun <yunjunz@outlook.com>

* Update contrib/stack/topsStack/stackSentinel.py

Co-authored-by: Zhang Yunjun <yunjunz@outlook.com>

* Update contrib/stack/topsStack/stackSentinel.py

Co-authored-by: Zhang Yunjun <yunjunz@outlook.com>

* Update contrib/stack/topsStack/stackSentinel.py

Co-authored-by: Zhang Yunjun <yunjunz@outlook.com>

* Update contrib/stack/topsStack/stackSentinel.py

Co-authored-by: Zhang Yunjun <yunjunz@outlook.com>

---------

Co-authored-by: Zhang Yunjun <yunjunz@outlook.com>

bjmarfito committed:
Fix for dloadOrbits.py to reflect new download URL and minor fix in fetchOrbit.py (#870)

bjmarfito committed:
Update README.md

Commit:
fix some incorrect pointer types as required by newer gcc compilers

Commit:
patch the time.h error for g++14

rtburns-jpl committed:
Merge pull request #897 from lijun99/modern-c

fix some incorrect pointer types as required by newer gcc compilers

rtburns-jpl committed:
Remove extraneous tab chars from topsStack readme

rtburns-jpl committed:
Merge pull request #886 from bjmarfito/readme_fix

Update README.md for topsStack ionospheric phase screen module

bjmarfito committed:
Bugfix plot ion topsStack (#885)

* Update plotIonDates.py

* Update plotIonPairs.py

rtburns-jpl committed:
Merge pull request #709 from yunjunz/bugfix_pycuampcor

bugfix in `cuDenseOffsets.py --full/out-geom` for `topsApp` product

yjzhenglamarmota committed:
Merge branch 'ersStack' of https://github.com/yjzhenglamarmota/isce2 into ersStack

yjzhenglamarmota committed:
Removed commented code as suggested by hfattahi.
